crates/codegen/parser_templates/src/shared/language.rs

```diff
@@ -176,3 +176,117 @@ pub(crate) fn render_error_report(
         .trim()
         .to_string();
 }
+
+#[allow(dead_code)]
+fn call_scanner<L, F>(
+    language: &L,
+    input: &str,
+    scanner: F,
+    kind: TokenKind,
+    error_message: &str,
+) -> Option<ParserOutput>
+where
+    F: Fn(&L, &mut Stream) -> bool,
+{
+    let mut stream = Stream::new(input);
+    Some(
+        if scanner(language, &mut stream) && stream.peek().is_none() {
+            ParserOutput {
+                parse_tree: Some(cst::Node::token(
+                    kind,
+                    Range {
+                        start: 0,
+                        end: stream.position(),
+                    },
+                    None,
+                    None,
+                )),
+                errors: vec![],
+            }
+        } else {
+            ParserOutput {
+                parse_tree: None,
+                errors: vec![ParseError::new(stream.position(), error_message)],
+            }
+        },
+    )
+}
+
+#[allow(dead_code)]
+fn call_maybe_scanner<L, F>(
```

Comment:
Nit: Thoughts on using `try_call_scanner()` vs `call_maybe_scanner()`?
I noticed `try_*` is often used with `Option<>`al operations.

Reply:
nice, done